### PR TITLE
Fix patch skip + emulation on ARM64

### LINKED ISSUE #40634: Arm64 breakpoint causes floating corruption
When adding a regression test for https://github.com/dotnet/coreclr/pull/28033, @hoyosjs reported the test failed on 5.0 linux-arm64, but not on 3.1 linux-arm64.

### PR DESCRIPTION
**Pull request by hoyosjs**
Fixes #40634

All the pending for cleanup for 6.0 is tracked in #40940.

### CODE REVIEW

File: src/coreclr/src/debug/ee/arm64/arm64walker.cpp [hunk 1]
```diff
@@ -187,81 +184,55 @@ BYTE*  NativeWalker::SetupOrSimulateInstructionForPatchSkip(T_CONTEXT * context,
...
-                    break;
-                case 3:
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate opcode: %x  as PRFM ,but do nothing \n", opcode));
-
-                    break;
-                default:
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate Unknown opcode: %x [LDR(literal)]  \n", opcode));
-                    _ASSERTE(!("Arm64Walker::Simulated Unknown opcode"));
-
+                    SimdRegContents.Low = GetMem(ip, 4, true);
```
Comment (sdmaclea) [inline src/coreclr/src/debug/ee/arm64/arm64walker.cpp:222]: This needs to be sign extended.  Lines 241-244 above.

**Review by sdmaclea — APPROVED** [review]
LGTM except missing sign extension code.

File: src/coreclr/src/debug/ee/arm64/arm64walker.cpp (hunk 1 above)
Comment (hoyosjs) [reply src/coreclr/src/debug/ee/arm64/arm64walker.cpp:222]: That's what the true is for. Maybe I should add a comment.

**Comment by hoyosjs** [comment]
- The latest commit only adds comments.
- The previous commit builds correctly (this is a win-arm64 change only).
- There's no tests in this repo that would exercise this code path. I've run them against Visual Studio and the behavior is as expected.

Merging for RC1

**Pull request merged**